src/pico/pico-screens/screens/ssd1306_70_40_i2c.c

```diff
@@ -0,0 +1,144 @@
+#include "ssd1306_70_40_i2c.h"
+#include "pico/ssd1306_i2c.h"
+#include "shared.h"
+
+
+
+    // .spi      = PICO_DEFAULT_SPI_INSTANCE,
+    // .gpio_din = PICO_DEFAULT_SPI_TX_PIN,
+    // .gpio_clk = PICO_DEFAULT_SPI_SCK_PIN,
+    // .gpio_cs  = PICO_DEFAULT_SPI_CSN_PIN,
+    // .gpio_dc  = 20,
+    // .gpio_rst = 21,
+    // .gpio_bl  = 22,
+// #define START_Y 24
+
+// +4 to center on _72_ pixel width screen
+// I counted, it's 72!
+#define START_X (4)
+
+static uint8_t buf[SSD1306_BUF_LEN+1];
+// for greyscale
+static uint8_t second_buf[SSD1306_BUF_LEN+1];
```

Comment:
Now we get to talk about greyscale, which I'm still working on.

The central idea behind greyscale on a monochrome display is simply that you display 2,3, or maybe 4 frames at 2x, 3x, or 4x the speed you were before. If a pixel is white it'll be on in all the frames, if it's black it'll be off, and if it's grey it'll be on in some frames and off in others. Persistence Of Vision kicks in and makes it all look like one image.

The problem with this, though, is you'll interrupt the screen mid-draw, causing tearing.

The central idea behind uGrey and Thumby's greyscale mod, is that you can abuse the multiplexing command and use dead reckoning to make it so you only interrupt the screen controller when it's drawing off-screen. That way you don't get any tearing.

The problem with _my_ implementation of this is it's not on a separate core, as I haven't attempted to unwind the multithreading logic yet to see if it would work. The core that draws the frames also does a lot of other stuff, and it does some particularly harder stuff when the game is running, resulting in flickering.

I'm going to attempt to get greyscale working and working well, but it may have to be its own branch. It might require significant changes that would impact the performance of other screens - like decoupling frame rendering from screen rendering again.